### PR TITLE
EPMDJ-3219 Added a delay in scanning classes

### PR DESCRIPTION
**Pull request by ViktoriiaLebedeva**

### CODE REVIEW

File: src/commonNative/kotlin/com/epam/drill/util/StringProperty.kt [hunk 1]
```diff
@@ -28,6 +28,10 @@ class StringProperty(val map: Map<String, String>) : NamedValueDecoder() {
         return map.getValue(tag).toBoolean()
     }
 
+    override fun decodeTaggedLong(tag: String): Long {
```
Comment (viktorbll) [inline src/commonNative/kotlin/com/epam/drill/util/StringProperty.kt:31]: There are no usages of this method in the request.

File: src/commonNative/kotlin/com/epam/drill/core/callbacks/classloading/ClassLoadHook.kt [hunk 2]
```diff
@@ -38,7 +38,7 @@ fun classLoadEvent(
         }
         val classReader = ClassReader(classBytes)
         val transformers = mutableListOf<() -> ByteArray?>()
-        if (!state.allWebAppsInitialized() && Transformer.servletListener in classReader.interfaces)
+        if (Transformer.servletListener in classReader.interfaces)
```
Comment (viktorbll) [inline src/commonNative/kotlin/com/epam/drill/core/callbacks/classloading/ClassLoadHook.kt:41]: Transformation of Servlet Context Listeners should be optional.

File: src/commonNative/kotlin/com/epam/drill/agent/State.kt [hunk 3]
```diff
@@ -18,6 +19,21 @@ var isAsyncApp: Boolean
         _isAsyncApp.value = value.freeze()
     }
 
+private val _classScanDelay = AtomicReference(0L).freeze()
+
+var classScanDelay: Long
```
Comment (viktorbll) [inline src/commonNative/kotlin/com/epam/drill/agent/State.kt:24]: Too much atomic variables. These age just configs and should be placed inside a config class.

File: src/commonNative/kotlin/com/epam/drill/agent/State.kt [hunk 4]
```diff
@@ -18,6 +19,21 @@ var isAsyncApp: Boolean
...
+    get() = _classScanDelay.value
+    set(value) {
+        _classScanDelay.value = value.freeze()
+    }
+
+val _latch = AtomicReference<Job?>(null).freeze()
```
Comment (viktorbll) [inline src/commonNative/kotlin/com/epam/drill/agent/State.kt:30]: This latch just complicated the code. The delay can be implemented inside "getClassesByConfig" callback.

**Review by viktorbll — CHANGES_REQUESTED** [review]

File: src/commonNative/kotlin/com/epam/drill/util/StringProperty.kt (hunk 1 above)
Comment (IgorKey) [reply src/commonNative/kotlin/com/epam/drill/util/StringProperty.kt:31]: This is required for parsing the agent's arguments. Previously there were only strings and Boolean, now there is a long for duration

**Review by viktorbll — APPROVED** [review]

**Pull request merged**